test/SlowTests/Issues/RavenDB_19239.cs

```diff
@@ -0,0 +1,125 @@
+using System.Linq;
+using FastTests;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+using Raven.Client.Exceptions;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_19239 : RavenTestBase
+{
+    public RavenDB_19239(ITestOutputHelper output) : base(output)
+    {
+    }
+    
+    [RavenFact(RavenTestCategory.Indexes)]
+    public void TestLockedError()
+    {
+        using (var store = GetDocumentStore())
+        {
+            using (var session = store.OpenSession())
+            {
+                var dto1 = new Dto() { Name = "Name1" };
+                var dto2 = new Dto() { Name = "Name2" };
+
+                session.Store(dto1);
+                session.Store(dto2);
+
+                session.SaveChanges();
+                
+                var index = new DummyIndex();
+
+                index.Execute(store);
+                
+                Indexes.WaitForIndexing(store);
+                
+                store.Maintenance.Send(new SetIndexesLockOperation("DummyIndex", IndexLockMode.LockedError));
+                
+                var ex = Assert.Throws<RavenException>(() => store.Maintenance.Send(new DeleteIndexOperation("DummyIndex")));
```

Comment:
Do we get `RavenException` or `IndexDeletionException` (which inherits from `RavenException`)?

Reply:
Here is what we get on PUT with `LockedError`:

https://github.com/ravendb/ravendb/blob/efea97e66cd52173fefa2eab79bdb55dbed9af1a/test/SlowTests/Issues/RavenDB_4110.cs#L62-L63

Reply:
I changed the message check so it's clear what type of exception we expect to throw

Reply:
Let's do the same as during index creation / update and throw the dedicated `IndexDeletionException`:

https://github.com/ravendb/ravendb/blob/2d9eabcf078a40e5eaabc1a013e81a363b5c46bc/src/Raven.Server/Documents/Indexes/AbstractIndexCreateController.cs#L117-L133